scripts/make-dist.sh

```diff
@@ -5,22 +5,21 @@
 set -e
 
 package_name=singularity
-package_version=`(git describe --match 'v[0-9]*' --always 2>/dev/null || cat VERSION 2>/dev/null || echo "") | sed -e "s/^v//;s/-/_/g;s/_/-/;s/_/./g"`
-package_version_short=`(git describe --abbrev=0 --match 'v[0-9]*' --always 2>/dev/null || cat VERSION 2>/dev/null || echo "") | sed -e "s/^v//;s/-/_/g;s/_/-/;s/_/./g"`
-
-# Remove Dist tarball if it exists
-if [ -f $package_name-$package_version_short.tar.gz ]; then
-    rm -f $package_name-$package_version_short.tar.gz
-fi
+package_version_short="`sed -n 's/^Version: //p' singularity.spec`"
+tree_version="$package_version_short-`sed -n 's/^Release: \([^%]*\).*/\1/p' singularity.spec`"
 
 echo " DIST setup VERSION: $tree_version"
 echo $tree_version > VERSION
-git add VERSION
-# spec file needs to be at the root of the project
-cp dist/rpm/singularity.spec .
-git add singularity.spec
-echo " DIST create tarball: $package_name-$package_version_short.tar.gz"
-git archive --format=tar --prefix=$package_name/ `git stash create` -o $package_name-$package_version_short.tar
-git reset VERSION singularity.spec
-gzip $package_name-$package_version_short.tar
-
+rmfiles="VERSION"
+tarball="$package_name-$package_version_short.tar.gz"
+echo " DIST create tarball: $tarball"
+rm -f $tarball
+pathtop="$package_name"
+thisdir="`basename $PWD`"
+if [ "$thisdir" != "$pathtop" ]; then
+    ln -fs $thisdir ../$pathtop
```

Comment:
Is that symlink the controversy you describe in your comment? Cause It breaks your build in this PR.

For the tarball directory name including the version number, I don't mind doing it. Unless @jmstover has reservations. Let's make sure we document how to prepare a source tree that is easy for users to unpack and build.

Reply:
That's not the controversy, the controversy was over whether or not to include the version number in the top level directory name.

There were two problems with the travis build.  The problem that caused it to fail was that the parent directory is not writable and the current directory is not named "singularity".  Instead I will make the symlink in the current directory.  The second problem was that it was trying to read from dist/rpm/singularity.spec; that would have caused a failure later because of missing packages.

Reply:
Ok Thanks for helping us wrapping that one up !